### PR TITLE
Restructure the survey response serializers.

### PR DESCRIPTION
**Pull request by maxpeterson**
The get and post format now match each other and multiple validation
errors will be included.

### CODE REVIEW

**Comment by maxpeterson** [comment]
@incuna/backend please review. JS updates to follow.

File: api-description/forms/pk/respond/user_id/post.json [hunk 1]
```diff
@@ -2,54 +2,30 @@
...
-            },
-            {
-                "fieldset": 3,
-                "answers": {
-                    "4": 0,
-                    "5": [1, 2]
-                }
-            }
-        ]
+        "1": {
```
Comment (maxpeterson) [inline api-description/forms/pk/respond/user_id/post.json]: @kanteo will this cause you issues?

Comment (KevinEtienne) [inline api-description/forms/pk/respond/user_id/post.json]: @cepages maybe?

**Comment by adam-thomas** [comment]
Looks like one of the frontend e2e tests needs updating:

```
Firefox 47.0.0 (Linux 0.0.0) FieldsetParserService parseModelToResponse object should be equal to the post data FAILED
	Expected [ Object({ fieldset: 1, answers: Object({ 1: 'Friends' }) }), Object({ fieldset: 2, answers: Object({ 3: 5 }) }), Object({ fieldset: 3, answers: Object({ 4: 0, 5: [ 1, 2 ] }) }) ] to equal undefined.
	require<["/home/travis/build/incuna/incuna-surveys/web/tests/fieldsets-parser-spec.js"]</</</<@/tmp/tests/fieldsets-parser-spec.js:110:0 <- /tmp/797da67b30eae5616519f9b6de9ee63f.browserify:607:13
Firefox 47.0.0 (Linux 0.0.0): Executed 17 of 30 (1 FAILED) (0 secs / 0.162 secs)
```

**Comment by maxpeterson** [comment]
@adam-thomas Is that comment after my "Update JS to match api changes." commit

**Comment by maxpeterson** [comment]
@incuna/backend / @incuna/js can this be merged?

File: surveys/tests/test_views_api.py [hunk 2]
```diff
@@ -142,15 +135,25 @@ def test_get(self):
 
         self.assertEqual(response.status_code, 200)
 
-        # Take the data to and from JSON to turn integer indices into strings.
-        data = json.loads(json.dumps(response.data))
+        # The data is asserted thoroughly in the serializer test, so we'll just check
+        # that it's the same data here and hasn't been tampered with.  That way, if the
+        # serializer is broken, only that test should fail, and this test will be
+        # unconcerned by changes to the serializer format.
```
Comment (adam-thomas) [inline surveys/tests/test_views_api.py]: :D

**Comment by adam-thomas** [comment]
@maxpeterson looks like it wasn't :)  Happy to merge as far as backend is concerned, nice work!

**Comment by cepages** [comment]
Sorry, I was to focus in my stuff. I'm ok with that :)

**Pull request merged**